From 2b374bffa17d8903283ba0b516ce3cdd1e4af88d Mon Sep 17 00:00:00 2001
From: Brian Liu <brian.liu@freescale.com>
Date: Thu, 15 Jul 2010 14:21:22 +0800
Subject: [PATCH 051/635] Fix compile error when upgrading to Froyo

Remove header include path for kernel_imx
Change header include path for libcamera
Change macro KEY_SUPPORTED_THUMBNAIL_SIZES to
KEY_SUPPORTED_JPEG_THUMBNAIL_SIZES
Link libcamera_client (instead of libui) into CameraHAL
Link libsurface into overlay_test

Fix compile error when upgrading to Froyo
---
 libcamera/Android.mk         | 2 +-
 libcamera/CameraHal.cpp      | 2 +-
 libcamera/CameraHal.h        | 2 +-
 libcopybit/Android.mk        | 1 -
 libgralloc/Android.mk        | 3 +--
 liboverlay/Android.mk        | 4 ++--
 liboverlay/overlays_test.cpp | 9 +++++----
 7 files changed, 11 insertions(+), 12 deletions(-)

diff --git a/libcamera/Android.mk b/libcamera/Android.mk
index d274f25..4f263a7 100755
--- a/libcamera/Android.mk
+++ b/libcamera/Android.mk
@@ -23,6 +23,7 @@ LOCAL_SRC_FILES:=               \
 LOCAL_CPPFLAGS += -DUSE_FSL_JPEG_ENC -DDUMP_CAPTURE_YUVxx -DCAPTURE_ONLY_TESTxx
 
 LOCAL_SHARED_LIBRARIES:= \
+    libcamera_client \
     libui \
     libutils \
     libcutils \
@@ -32,7 +33,6 @@ LOCAL_SHARED_LIBRARIES:= \
     libc
 
 LOCAL_C_INCLUDES += \
-	kernel_imx/include \
 	frameworks/base/include/binder \
 	frameworks/base/include/ui \
 	frameworks/base/camera/libcameraservice
diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 754106a..13d9719 100644
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -119,7 +119,7 @@ void CameraHal::initDefaultParameters()
     p.set(CameraParameters::KEY_SUPPORTED_PREVIEW_SIZES, CameraHal::supportedPreviewSizes);
     p.set(CameraParameters::KEY_SUPPORTED_PREVIEW_FORMATS, CameraParameters::PIXEL_FORMAT_YUV420SP);
     p.set(CameraParameters::KEY_SUPPORTED_PREVIEW_FRAME_RATES, CameraHal::supportedFPS);
-    p.set(CameraParameters::KEY_SUPPORTED_THUMBNAIL_SIZES, CameraHal::supprotedThumbnailSizes);
+    p.set(CameraParameters::KEY_SUPPORTED_JPEG_THUMBNAIL_SIZES, CameraHal::supprotedThumbnailSizes);
 
     memset(tmpBuffer, '\0', PARAM_BUFFER);
     strncat((char*) tmpBuffer, (const char*) CameraParameters::WHITE_BALANCE_AUTO, PARAM_BUFFER);
diff --git a/libcamera/CameraHal.h b/libcamera/CameraHal.h
index 931c131..03dac15 100644
--- a/libcamera/CameraHal.h
+++ b/libcamera/CameraHal.h
@@ -43,7 +43,7 @@
 #include <utils/threads.h>
 #include <binder/MemoryBase.h>
 #include <binder/MemoryHeapBase.h>
-#include <ui/CameraHardwareInterface.h>
+#include <camera/CameraHardwareInterface.h>
 #include <ui/Overlay.h>
 
 #define FB_DEVICE               "/dev/graphics/fb0"
diff --git a/libcopybit/Android.mk b/libcopybit/Android.mk
index 45bbbe2..c0d9f20 100755
--- a/libcopybit/Android.mk
+++ b/libcopybit/Android.mk
@@ -23,7 +23,6 @@ include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_C_INCLUDES += hardware/mx5x/libgralloc
-LOCAL_C_INCLUDES += kernel_imx/include
 LOCAL_C_INCLUDES += external/fsl-linux-amd-gpu/driver/include/private/C2D
 LOCAL_SHARED_LIBRARIES := liblog libcutils libc2d_z430
 
diff --git a/libgralloc/Android.mk b/libgralloc/Android.mk
index 6765821..f204cc7 100755
--- a/libgralloc/Android.mk
+++ b/libgralloc/Android.mk
@@ -21,8 +21,7 @@ include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SHARED_LIBRARIES := liblog libcutils libGLESv1_CM libipu
-LOCAL_C_INCLUDES += kernel_imx/include \
-		    external/fsl_imx_lib/ipu \
+LOCAL_C_INCLUDES += external/fsl_imx_lib/ipu
 
 LOCAL_SRC_FILES := 	\
 	allocator.cpp 	\
diff --git a/liboverlay/Android.mk b/liboverlay/Android.mk
index 0bfb917..7dbfec0 100755
--- a/liboverlay/Android.mk
+++ b/liboverlay/Android.mk
@@ -36,8 +36,7 @@ LOCAL_SRC_FILES := \
 		overlay_pmem.cpp
 
 LOCAL_C_INCLUDES := \
-	kernel_imx/include \
-	external/fsl_imx_lib/ipu \
+	external/fsl_imx_lib/ipu
 
 LOCAL_MODULE := overlay.$(TARGET_BOARD_PLATFORM)
 
@@ -54,6 +53,7 @@ LOCAL_SRC_FILES:= \
 LOCAL_SHARED_LIBRARIES := \
 	libcutils \
 	libutils \
+	libsurfaceflinger \
 	libui
 
 LOCAL_MODULE:= mxc-test-overlays
diff --git a/liboverlay/overlays_test.cpp b/liboverlay/overlays_test.cpp
index 2fc3582..6c53cc3 100755
--- a/liboverlay/overlays_test.cpp
+++ b/liboverlay/overlays_test.cpp
@@ -19,10 +19,11 @@
 #include <utils/Log.h>
 
 #include <ui/DisplayInfo.h>
-#include <ui/Surface.h>
-#include <ui/ISurface.h>
 #include <ui/Overlay.h>
-#include <ui/SurfaceComposerClient.h>
+#include <surfaceflinger/Surface.h>
+#include <surfaceflinger/ISurface.h>
+#include <surfaceflinger/ISurfaceComposer.h>
+#include <surfaceflinger/SurfaceComposerClient.h>
 
 #include <fcntl.h>
 #include <stdint.h>
@@ -593,7 +594,7 @@ void *overlay_instance_test(void *arg)
     // now request an overlay
     LOGI("createOverlay()");
     sp<OverlayRef> ref = isurface->createOverlay(overlay_param->w, overlay_param->h,
-                                                 overlay_param->format);
+                                                 overlay_param->format, 0);
     LOGI("createOverlay() AFTER");
     LOGI("new Overlay(ref) start");
     sp<Overlay> overlay = new Overlay(ref);
-- 
1.8.0

